packages/contracts/contracts/L1/AssetManager.sol

```diff
@@ -388,16 +402,21 @@ contract AssetManager is ISemver, IERC721Receiver, IAssetManager {
     /**
      * @inheritdoc IAssetManager
      */
-    function initUndelegate(address validator, uint128 shares) external {
+    function undelegate(address validator, uint128 shares) external {
         if (shares == 0) revert NotAllowedZeroInput();
         if (shares > _vaults[validator].kroDelegators[msg.sender].shares)
             revert InsufficientShare();
 
         uint128 assets = _convertToKroAssets(validator, shares);
         if (assets == 0) revert InsufficientAsset();
+        if (canUndelegateKroAt(validator, msg.sender) > block.timestamp)
```

Comment:
```suggestion
        if (canUndelegateKroAt(validator, msg.sender) > uint128(block.timestamp))
```

Reply:
I thought it's just ok not to wrap it with uint128, is it better to wrap it in gas usage?

Reply:
I missed the comment above, and agree that it will be okay not to wrap! I just checked about the gas, and not wrapping is slightly cheaper.